Silence warning from Eigen on Windows

On Windows I'm getting a huge number of warnings from Eigen, of the form:
```
eigen-3.4.0\include\eigen3\eigen\src/Core/CoreEvaluators.h(219): warning C4127: conditional expression is constant
eigen-3.4.0\include\eigen3\eigen\src/Core/CoreEvaluators.h(219): note: consider using 'if constexpr' statement instead
```
Those warnings don't add much and since everything in Eigen is in headers, they appears a huge number of times.

So I suggest removing them by changing:
https://github.com/gstlearn/gstlearn/blob/da060726e16bc4f3e1a99a7a064672e3628e62a7/cmake/cpp.cmake#L18-L25
to:
```CMake
add_compile_options(/bigobj /W4 /wd4251 /wd4244 /wd4127) # Except those warnings 
```

--- a/cmake/cpp.cmake
+++ b/cmake/cpp.cmake
@@ -19,7 +19,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 # https://cmake.org/cmake/help/latest/command/add_compile_options.html
 if (MSVC)
   # Warning level 4 (4 = maximum, 0 = none)
-  add_compile_options(/bigobj /W4 /wd4251 /wd4244) # Except those two warnings
+  add_compile_options(/bigobj /W4 /wd4251 /wd4244 /wd4127) # Except those two warnings
   # Silence MSVC warnings about unsafe C standard library functions
   add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
 else()

